### PR TITLE
Convert ConvertSwitchStatementToExpressionTests to the new test framework

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

**Comment by mavasani** [comment]
Tests seem to be failing because of the default severity of the diagnostic.

File: src/EditorFeatures/CSharpTest/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs [hunk 1]
```diff
@@ -4,36 +4,35 @@
...
-    public partial class ConvertSwitchStatementToExpressionTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    public class ConvertSwitchStatementToExpressionTests
     {
-        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
-            => (new ConvertSwitchStatementToExpressionDiagnosticAnalyzer(), new ConvertSwitchStatementToExpressionCodeFixProvider());
-
-        private Task TestInCSharp8(string actual, string expected)
-            => TestInRegularAndScriptAsync(actual, expected, parseOptions: TestOptions.Regular8);
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertSwitchStatementToExpression)]
+        public void TestStandardProperties()
```
Comment (mavasani) [inline src/EditorFeatures/CSharpTest/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs]: Can you file an issue on roslyn-analyzers to write an analyzer for flagging test files without this method? Probably should be disabled by default.

File: src/EditorFeatures/CSharpTest/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionFixAllTests.cs [hunk 2]
```diff
@@ -160,54 +164,59 @@ int M(int i, int j)
...
         };
     }
-}");
+}";
+
+            await new VerifyCS.Test
+            {
+                TestCode = input,
+                FixedCode = expected,
+                NumberOfFixAllIterations = 2,
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionFixAllTests.cs]: 📝 This always means the Fix All fails to fix _all_ (in one invocation). The `expected` output was achieved via execution of the Fix All operation twice in sequence.

File: src/EditorFeatures/CSharpTest/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs [hunk 3]
```diff
@@ -196,86 +195,91 @@ int M(int i)
...
-@"class Program
+            var code = @"class Program
 {
     void M(int i)
     {
-        [||]switch (i)
+        switch (i)
         {
-            case _:
+            case { }:
```
Comment (sharwell) [inline src/EditorFeatures/CSharpTest/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs]: 📝 This test arguably was never correct and likely could have been removed. However, based on context (other tests with related names) I decided to use another form in the spirit of the original.

**Comment by sharwell** [comment]
> Tests seem to be failing because of the default severity of the diagnostic.

The failure was caused because I lost dotnet/roslyn-sdk#462 when I rebased this branch to make a pull request. The change is now part of **master** so I'll re-run the tests and expect them to pass.

File: src/EditorFeatures/CSharpTest/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs (hunk 1 above)
Comment (sharwell) [reply src/EditorFeatures/CSharpTest/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs]: ➡️ Filed https://github.com/dotnet/roslyn-sdk/issues/467

File: src/EditorFeatures/CSharpTest/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs (hunk 3 above)
Comment (alrz) [reply src/EditorFeatures/CSharpTest/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs]: fwiw, I think it should have been `var _` - `case {}` doesn't behave like `default:` (tests for null)

Comment (sharwell) [reply src/EditorFeatures/CSharpTest/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs]: `case var _` is handled by `TestMissingOnDefaultBreak_03` below

Comment (sharwell) [reply src/EditorFeatures/CSharpTest/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs]: Also note that `i` is a non-nullable value type, so `{ }` and `var _` are equivalent.

**Pull request merged**